geocoding/geocode.py

```diff
@@ -1,65 +1,81 @@
-    count = 0
-    total_count = 0
-    lats, longs, geocoded = [], [], []
-    print(f"{sheet_name} sheet has {len(addresses)} rows")
-    print()
-
-    for address in addresses:
-        try:
-            location = gmaps.geocode(address, components={"country": "IN"})
-            geocoded.append(True)
-        except:
-            geocoded.append(False)
-
-        try:
-            location = location[0]['geometry']['location']
-            lats.append(location['lat'])
-            longs.append(location['lng'])
-        except:
-            count+=1
-            lats.append(None)
-            longs.append(None)
-            
-        total_count += 1
-        
-        if total_count%50==0:
-            print(f"Geocoded {total_count} rows in {sheet_name}")
-
-    df['Latitude'] = lats
-    df['Longitude'] = longs
-    df['Geocoded'] = geocoded
-
-    print()
-    print(f"Saving {sheet_name} with geocoded address")
-    print(f"Unable to geocode - {count}")
-    total_none_count+=count
-    total_xlsx_count+=total_count
-    print()
-    df.to_csv(f"Sheets/{sheet_name}.csv", index=False)
-
-print(f"Total Addreses - {total_xlsx_count}")
-print(f"Total Addreses unable to geocode - {total_none_count}")
\ No newline at end of file
+    Args:
+        excel_sheet (str): Path to the sheet that is to be geocoded.
+        api_key (str): API key for Google Map Geocoding API
+        use_column (str, optional): The column that is to be used as input to API while geocoding. Defaults to "Address".
+    """
+    make_dir()
+    xlsx_name = pathlib.Path.cwd() / excel_sheet
+    temp = pd.ExcelFile(xlsx_name)
+    sheet_names = temp.sheet_names
+
+    gmaps = googlemaps.Client(key=api_key)
+
+    total_xlsx_count = 0
+    total_none_count = 0
+
+    for sheet_name in sheet_names:
+        df = pd.read_excel(xlsx_name, sheet_name=sheet_name)
+        addresses = df[use_column].tolist()
+        count = 0
+        total_count = 0
+        lats, longs, geocoded = [], [], []
+        print(f"{sheet_name} sheet has {len(addresses)} rows")
+        print()
+
+        for address in tqdm(addresses):
+            try:
+                location = gmaps.geocode(address, components={"country": "IN"})
+                geocoded.append(True)
+            except:
+                geocoded.append(False)
+
+            try:
+                location = location[0]["geometry"]["location"]
+                lats.append(location["lat"])
+                longs.append(location["lng"])
+            except:
+                count += 1
+                lats.append(None)
+                longs.append(None)
+            total_count += 1
+
+        df["Latitude"] = lats
+        df["Longitude"] = longs
+        df["Geocoded"] = geocoded
+
+        print()
+        print(f"Saving {sheet_name} with geocoded address")
+        print(f"Unable to geocode - {count}")
+        total_none_count += count
+        total_xlsx_count += total_count
+        print()
```

Comment:
Blank print statements?

Reply:
These should probably be logging statements going to `stdout` or a file or both? I believe the blank print statements were for debugging and are no longer necessary?

Reply:
Added logging rather than `print` statements.